### PR TITLE
build(fsevents+bower) add script to remove mac dependency + run bower after npm install

### PR DESCRIPTION
**Pull request by gary-b**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [n/a] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
(from #12443)
There are additional steps required to build angular not included in the developer guide, some of which could be scripted.

a) On Windows / Linux npm install will fail because fsevents requires a mac.

b) On Windows build.sh fails because it tries to create a symlink to the polymer directory and it doesn't exist. This is because the dependency is managed by bower, and bower install is never ran. (The script does not fail on Linux when the directory is missing.)

**What is the new behavior?**
a) Added a script to remove the fsevents dependency which is to be run by developers when not on a mac. Updated dev guide to advise of this. (Better solutions welcome...)
b) Now have bower install run automatically when npm install is ran (using the packages.json postinstall step).

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

**Other information**:

### CODE REVIEW

**Comment by vicb** [comment]
commit comment syntax: `build: ...`

**Comment by gary-b** [comment]
Updated the commit comment.

**Comment by vicb** [comment]
thanks !

File: remove-mac-dependencies.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/* The fsevents package is required when building on Mac, but will cause npm install to
```
Comment (vicb) [inline remove-mac-dependencies.js]: please add the copyright header on top of this file

**Comment by gary-b** [comment]
Copyright message added.

**Comment by DzmitryShylovich** [comment]
@gary-b is this still valid? `fsevents` is optional now see https://github.com/angular/angular/pull/13945

**Comment by IgorMinar** [comment]
this is now obsolete. sorry about dropping the ball on this one. :-(

**Pull request closed without merging**